include/h2o/memory.h

```diff
@@ -257,14 +257,23 @@ void h2o_buffer__do_free(h2o_buffer_t *buffer);
  */
 static void h2o_buffer_dispose(h2o_buffer_t **buffer);
 /**
- * allocates a buffer.
+ * allocates a buffer. aborts on allocation failure.
  * @param inbuf - pointer to a pointer pointing to the structure (set *inbuf to NULL to allocate a new buffer)
  * @param min_guarantee minimum number of bytes to reserve
  * @return buffer to which the next data should be stored
  * @note When called against a new buffer, the function returns a buffer twice the size of requested guarantee.  The function uses
  * exponential backoff for already-allocated buffers.
  */
 h2o_iovec_t h2o_buffer_reserve(h2o_buffer_t **inbuf, size_t min_guarantee);
+/**
+ * allocates a buffer.
+ * @param inbuf - pointer to a pointer pointing to the structure (set *inbuf to NULL to allocate a new buffer)
+ * @param min_guarantee minimum number of bytes to reserve
+ * @return buffer to which the next data should be stored
+ * @note When called against a new buffer, the function returns a buffer twice the size of requested guarantee.  The function uses
+ * exponential backoff for already-allocated buffers.
+ */
```

Comment:
How about stating the differences rather than repeating the description of `h2o_buffer_reserve`? It might be a good idea to only explain what is being returned.

Reply:
Sorry about this. I admit that this was very lazy. I'll make an appropriate edit.

Reply:
I decided to leave alone the documentation of `h2o_buffer_try_reserve()` because the only change is the function name.  Instead, edited the documentation of the "new" `h2o_buffer_reserve()`, such that:

```
/**
 * allocates a buffer with h2o_buffer_try_reserve. aborts on allocation failure.
 * @return buffer to which the next data should be stored
 */
h2o_iovec_t h2o_buffer_reserve(h2o_buffer_t **inbuf, size_t min_guarantee);
```

This is in: daeaf9006c3a096675a3ef0ec3a11c0baee9f843